api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupTxPvls.java

```diff
@@ -0,0 +1,89 @@
+package org.openmrs.module.eptsreports.reporting.reports;
+
+import org.openmrs.module.eptsreports.reporting.library.datasets.TxPvlsDataset;
+import org.openmrs.module.eptsreports.reporting.reports.manager.EptsDataExportManager;
+import org.openmrs.module.reporting.ReportingConstants;
+import org.openmrs.module.reporting.evaluation.parameter.Parameter;
+import org.openmrs.module.reporting.evaluation.parameter.ParameterizableUtil;
+import org.openmrs.module.reporting.report.ReportDesign;
+import org.openmrs.module.reporting.report.definition.ReportDefinition;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+
+@Component
+public class SetupTxPvls extends EptsDataExportManager {
+	
+	@Autowired
+	private TxPvlsDataset txPvlsDataset;
+	
+	@Override
+	public String getExcelDesignUuid() {
+		return "e57f0ec4-e287-11e8-acd5-e311f10b26ed";
+	}
+	
+	@Override
+	public String getUuid() {
+		return "f3af9590-e287-11e8-bba1-e73bca3486f8";
+	}
+	
+	@Override
+	public String getName() {
+		return "Tx_Pvls Report";
+	}
+	
+	@Override
+	public String getDescription() {
+		return "Percentage of ART patients with a viral load result documented in the medical record and/or laboratory information systems (LIS) within the past 12 months with a suppressed viral load (<1000 copies/ml)";
+	}
+	
+	@Override
+	public ReportDefinition constructReportDefinition() {
+		ReportDefinition rd = new ReportDefinition();
+		rd.setUuid(getUuid());
+		rd.setName(getName());
+		rd.setDescription(getDescription());
+		rd.setParameters(getParameters());
+		
+		rd.addDataSetDefinition(txPvlsDataset.constructTxPvlsDatset(),
+		    ParameterizableUtil.createParameterMappings("endDate=${endDate},startDate=${startDate}"));
+		
+		return rd;
+	}
+	
+	@Override
+	public String getVersion() {
+		return "1.0-SNAPSHOT";
+	}
+	
+	@Override
+	public List<Parameter> getParameters() {
+		List<Parameter> parameters = new ArrayList<Parameter>();
+		parameters.add(ReportingConstants.START_DATE_PARAMETER);
+		parameters.add(ReportingConstants.END_DATE_PARAMETER);
+		return parameters;
+	}
+	
+	@Override
+	public List<ReportDesign> constructReportDesigns(ReportDefinition reportDefinition) {
+		ReportDesign reportDesign = null;
+		try {
+			reportDesign = createRowPerPatientXlsOverviewReportDesign(reportDefinition, "TxPvls.xls", "TxPvls.xls_", null);
```

Comment:
#createRowPerPatientXlsOverviewReportDesign will now take a string parameter to initialize the report design uuid. The call will look like this:
_createXlsReportDesign(reportDefinition, "TXCURR.xls", "TXCURR.xls_", getExcelDesignUuid(), null);_

This is from our previous discussion where @ningosi  mentioned that the design resource was not being assigned the correct uuid. The proper way to make this change would have probably been to add an overload for the method and deprecate this one, but as we are in the initial development phase of the module and the method is not used in many places, I just replaced it with this one which takes the uuid parameter and [set it](https://github.com/esaude/openmrs-module-eptsreports/blob/MEPTS-23/api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/manager/EptsDataExportManager.java#L99) as the design resource uuid. @nthfloor -You'll get merger conflicts and you can refer to this comments.

Reply:
Sure, after merge, we will revert to common used standard approach